c7n/resources/quotas.py

```diff
@@ -0,0 +1,481 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import re
+import math
+
+from concurrent.futures import as_completed
+from datetime import timedelta, datetime
+from statistics import mean
+
+from c7n.actions import Action
+from c7n.exceptions import PolicyExecutionError
+from c7n.filters import Filter, ValueFilter
+from c7n.filters.metrics import MetricsFilter
+from c7n.filters.related import RelatedResourceFilter
+from c7n.manager import resources
+from c7n.query import QueryResourceManager, TypeInfo
+from c7n.utils import local_session, type_schema, get_retry
+
+
+@resources.register('service-quota-request')
+class ServiceQuotaRequest(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'service-quotas'
+        permission_prefix = 'servicequotas'
+        enum_spec = ('list_requested_service_quota_change_history', 'RequestedQuotas', None)
+        name = id = 'Id'
+        # Service Quota Requests dont actually have arns, but we need to set an arn_type here
+        # to ensure that the tests pass resource type validation
+        arn_type = 'request'
+
+
+@resources.register('service-quota')
+class ServiceQuota(QueryResourceManager):
+
+    batch_size = 100
+
+    class resource_type(TypeInfo):
+        service = 'service-quotas'
+        permission_prefix = 'servicequotas'
+        enum_spec = ('list_services', 'Services', None)
+        id = 'QuotaCode'
+        arn = 'QuotaArn'
+        name = 'QuotaName'
+        metrics_namespace = 'AWS/Usage'
+
+    def augment(self, resources):
+        client = local_session(self.session_factory).client('service-quotas')
+        retry = get_retry(('TooManyRequestsException',))
+
+        def get_quotas(client, s):
+            quotas = {}
+            token = None
+            kwargs = {
+                'ServiceCode': s['ServiceCode'],
+                'MaxResults': self.batch_size
+            }
+            while True:
+                if token:
```

Comment:
is the lack of bultin pagination the only reason this doesn't work as a detail_spec ? ie. is the set logic/uniqueness here needed?